lib/bumbler/hooks.rb

```diff
@@ -19,30 +13,67 @@ class << self
 
       # Inject our custom handling of require into the Kernel.
       def hook_require!
-        @hooking_require = true
+        hook_instance_require!
+        hook_singleton_require!
+      end
+
+      def hook_instance_require!
+        @hooking_instance_require = true
+
+        ::Kernel.module_eval do
+          orig_instance_require = instance_method(:require)
+          define_method(:require) do |path, *args|
+            ::Bumbler::Hooks.handle_require(path) do
+              orig_instance_require.bind(self).call(path, *args)
+            end
+          end
+          private :require
+        end
+
+        @hooking_instance_require = nil
+      end
+
+      def hook_singleton_require!
+        @hooking_singleton_require = true
 
-        # There are two independent require methods.  Joy!
-        ::Kernel.prepend RequireLogger
-        (class << ::Kernel; self; end).prepend RequireLogger
+        ::Kernel.module_eval do
+          class << self
+            orig_public_require = Kernel.public_method(:require)
+            define_method(:require) do |path, *args|
+              ::Bumbler::Hooks.handle_require(path) do
+                orig_public_require.call(path, *args)
+              end
+            end
+          end
+        end
 
-        @hooking_require = nil
+        @hooking_singleton_require = nil
```

Comment:
@grosser This isn't working perfectly but I found out the bug with bootsnap, bumbler was only hooking the instance require and bootsnap was hooking singleton singleton after the instance, so only the singleton method was getting overridden.

Reply:
For a variety of reasons I don't think there's a way to make the prepend method of hooking work with all the other ways require gets hooked unfortunately.

Reply:
that's alright, as long as the tests fail when someone tries to bring back prepend / or we have reproduction docs nobody should fall into this rabbit hole again :)